### PR TITLE
Introduce CFG_CC_OPT_LEVEL and deprecate CFG_CC_OPTIMIZE_FOR_SIZE

### PR DESCRIPTION
**Pull request by jforissier**
The influence on the performance of the C optimization flag (-O) can be
significant as shown by the output of "time xtest":
```
      | QEMUv8         | HiKey960 (A73 cores only)
 -----+----------------+--------------------------
  -O0 | 2m 54s, 2m 49s | 42.28s, 42.07s
  -Os | 2m 03s, 2m 03s | 25.57s, 25.60s
  -O2 | 1m 36s, 1m 35s | 24.01s, 23.93s
```
This commit introduces CFG_CC_OPT_LEVEL (default: s) which may be set to
0, s, 2 or any value accepted by the compiler. This gives better
flexibility to chose the best level depending on the use case.

Signed-off-by: Jerome Forissier <jerome@forissier.org>

### CODE REVIEW

File: core/arch/arm/arm.mk [hunk 1]
```diff
@@ -140,17 +140,18 @@ arm64-platform-cflags-generic := -mstrict-align $(call cc-option,-mno-outline-at
...
+$(call force,CFG_CC_OPT_LEVEL,0)
 $(call force,CFG_DEBUG_INFO,y)
 endif
-
-CFG_CC_OPTIMIZE_FOR_SIZE ?= y
-ifeq ($(CFG_CC_OPTIMIZE_FOR_SIZE),y)
-platform-cflags-optimization ?= -Os
-else
-platform-cflags-optimization ?= -O0
+ifeq ($(CFG_CC_OPTIMIZE_FOR_SIZE),n)
```
Comment (etienne-lms) [inline core/arch/arm/arm.mk:146]: Backward compat with `CFG_CC_OPTIMIZE_FOR_SIZE=n` but not with `CFG_CC_OPTIMIZE_FOR_SIZE=y?`

Comment (jforissier) [reply core/arch/arm/arm.mk:146]: `-Os` is the default value so nothing special is needed I think?

Comment (etienne-lms) [reply core/arch/arm/arm.mk:146]: Ok, fine as is. Thanks for this.
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by jforissier** [comment]
I noticed that `CFG_CC_OPT_LEVEL=3` doesn't work. If anyone feels like debugging optimized code... :wink:

**Pull request merged**